device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceIO.java

```diff
@@ -215,26 +167,14 @@ private void commonOpenSetup()
      */
     public void close() throws IOException
     {
-        /* Codes_SRS_DEVICE_IO_21_017: [The close shall finish all ongoing tasks.] */
-        /* Codes_SRS_DEVICE_IO_21_018: [The close shall cancel all recurring tasks.] */
-        if (taskScheduler != null)
-        {
-            this.taskScheduler.shutdown();
-        }
```

Comment:
Where do you shutdown the taskScheduler ?

Reply:
When the method onTransportConnectionClosed() is called. That method is called either when a customer calls closeNow() manually, or when the Transport layer loses connection and cannot recover. That way the scheduler shuts down no matter how the SDK gets to a DISCONNECTED state.